doc/changelog.md

```diff
@@ -63,104 +108,96 @@
+    the signature `qjit(fn, *args, **kwargs)`, where `fn` is the function to be compiled.
+* The compiler driver diagnostic output has been improved, and now includes failing IR as well as
+  names of failing passes.
+  [(#349)](https://github.com/PennyLaneAI/catalyst/pull/349)
 
-* Include the "Catalyst" utility dialect in our MLIR C-API.
-  [(#345)](https://github.com/PennyLaneAI/catalyst/pull/345)
+* The scatter operation in the Catalyst dialect now uses a SCF for loop to avoid ballooning
+  the compiled code.
+  [(#307)](https://github.com/PennyLaneAI/catalyst/pull/307)
+
+* The `CopyGlobalMemRefPass` pass of our MLIR processing pipeline now supports
+  dynamically shaped arrays.
+  [(#348)](https://github.com/PennyLaneAI/catalyst/pull/348)
 
 <h3>Breaking changes</h3>
 
 * The axis ordering for `catalyst.jacobian` is updated to match `jax.jacobian`. Assume we have
-  parameters of shape `[a,b]` and results of shape `[c,d]`. The jacobian would get the shape
-  `[c,d,a,b]` instead of `[a,b,c,d]`.
+  parameters of shape `[a,b]` and results of shape `[c,d]`, the returned Jacobian will now have
+  shape `[c, d, a, b]` instead of `[a, b, c, d]`.
   [(#283)](https://github.com/PennyLaneAI/catalyst/pull/283)
 
 <h3>Bug fixes</h3>
 
-* Enable AutoGraph to convert functions even when they are invoked through functional wrappers such
-  as `adjoint`, `ctrl`, `grad`, `jacobian`, etc.
-  [(#336)](https://github.com/PennyLaneAI/catalyst/pull/336)
-
-  The following should now succeed:
+* Fixes the issue with missing `CFP_t` in `StateVectorLQubitDynamic` when building against the
+  master branch of PennyLane-Lightning. This issue was introduced in [(#499)](https://github.com/PennyLaneAI/pennylane-lightning/pull/499).
+  [(#322)](https://github.com/PennyLaneAI/catalyst/pull/322)
 
-  ```python
-  def inner(n):
-    for i in range(n):
-      qml.T(i)
+* The `requirements.txt` file to build Catalyst from source has been updated with a minimum pip
+  version, `>=22.3`. Previous versions of pip are unable to perform editable installs when the
+  system-wide site-packages are read-only, even when the `--user` flag is provided.
+  [(#311)](https://github.com/PennyLaneAI/catalyst/pull/311)
 
-  @qjit(autograph=True)
-  @qml.qnode(dev)
-  def f(n: int):
-      adjoint(inner)(n)
-      return qml.state()
-  ```
+* The frontend has been updated to make it compatible with PennyLane `MeasurementProcess` objects
+  now being PyTrees in PennyLane version 0.33.
+  [(#315)](https://github.com/PennyLaneAI/catalyst/pull/315)
 
-* Fixes the issue with missing `CFP_t` in `StateVectorLQubitDynamic` when building against the
-  master branch of PennyLane-Lightning. This issue was introduced in
-  [PR 499](https://github.com/PennyLaneAI/pennylane-lightning/pull/499).
-  [(#322)](https://github.com/PennyLaneAI/catalyst/pull/322)
+* The Catalyst utility dialect is now included in the Catalyst MLIR C-API.
+  [(#345)](https://github.com/PennyLaneAI/catalyst/pull/345)
```

Comment:
This is an improvement rather than a bug fix :)

Reply:
Oops. Definitely move it up in that case!